File: Bug_Bounty/get_html_title.py
import argparse
import requests
from bs4 import BeautifulSoup

def get_html_title(url):
    try:
        response = requests.get(url)
        response.raise_for_status()
        soup = BeautifulSoup(response.content, 'html.parser')
        title = soup.title.string.strip()
        return title
    except (requests.RequestException, AttributeError):
        return None

def get_titles_from_file(input_file, output_file, verbose):
    with open(input_file, 'r') as file:
        with open(output_file, 'w') as output:
            for line in file:
                url = line.strip()
                title = get_html_title(url)
                if title:
                    result = f"{url}: {title}\n"
                else:
                    result = f"{url}: Title not found\n"

                if verbose:
                    print(result)
                
                output.write(result)

# Create argument parser
parser = argparse.ArgumentParser(description='Fetch HTML titles from a list of websites.')
parser.add_argument('input_file', help='path to the input file')
parser.add_argument('output_file', help='path to the output file')
parser.add_argument('-v', '--verbose', action='store_true', help='display output in real-time')

# Print help menu if no arguments are provided
if len(sys.argv) == 1:
    parser.print_help(sys.stderr)
    sys.exit(1)

# Parse the command-line arguments
args = parser.parse_args()

# Execute the script
get_titles_from_file(args.input_file, args.output_file, args.verbose)

File: Bug_Bounty/get_html_title.go
package main

import (
	"bufio"
	"crypto/tls"
	"flag"
	"fmt"
	"log"
	"net/http"
	"os"

	"golang.org/x/net/html"
)

func getHTMLTitle(url string) (string, error) {
	// Ignore SSL errors
	tr := &http.Transport{
		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
	}
	client := &http.Client{Transport: tr}

	resp, err := client.Get(url)
	if err != nil {
		return "", err
	}
	defer resp.Body.Close()

	doc, err := html.Parse(resp.Body)
	if err != nil {
		return "", err
	}

	var extractTitle func(*html.Node) string
	extractTitle = func(n *html.Node) string {
		if n.Type == html.ElementNode && n.Data == "title" && n.FirstChild != nil {
			return n.FirstChild.Data
		}
		for c := n.FirstChild; c != nil; c = c.NextSibling {
			title := extractTitle(c)
			if title != "" {
				return title
			}
		}
		return ""
	}

	title := extractTitle(doc)
	if title == "" {
		return "", fmt.Errorf("Title not found")
	}
	return title, nil
}

func getTitlesFromFile(inputFile, outputFile string, verbose bool) error {
	file, err := os.Open(inputFile)
	if err != nil {
		return err
	}
	defer file.Close()

	outFile, err := os.Create(outputFile)
	if err != nil {
		return err
	}
	defer outFile.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		url := scanner.Text()
		title, err := getHTMLTitle(url)
		if err != nil {
			if verbose {
				fmt.Printf("%s: %v\n", url, err)
			}
			outFile.WriteString(fmt.Sprintf("%s: %v\n", url, err))
		} else {
			if verbose {
				fmt.Printf("%s: %s\n", url, title)
			}
			outFile.WriteString(fmt.Sprintf("%s: %s\n", url, title))
		}
	}

	if err := scanner.Err(); err != nil {
		return err
	}
	return nil
}

func main() {
	inputFile := flag.String("input", "", "path to the input file")
	outputFile := flag.String("output", "", "path to the output file")
	verbose := flag.Bool("v", false, "display output in real-time")

	flag.Parse()

	if *inputFile == "" || *outputFile == "" {
		flag.PrintDefaults()
		os.Exit(1)
	}

	err := getTitlesFromFile(*inputFile, *outputFile, *verbose)
	if err != nil {
		log.Fatal(err)
	}
}
